configure.ac

```diff
@@ -246,6 +246,18 @@ AC_ARG_ENABLE([memory-checking],
 AC_MSG_CHECKING([whether to enable memory checking])
 AC_MSG_RESULT([$enable_memory_checking])
 
+AC_ARG_ENABLE([valgrind],
+    [AS_HELP_STRING([--enable-valgrind], [add valgrind extensions to GASMAN])],
+    [AC_DEFINE([MEMORY_CANARY], [1], [define if build with valgrind extensions])],
+    [enable_valgrind=no]
+    )
+AC_MSG_CHECKING([whether to add valgrind extensions to GASMAN])
+AC_MSG_RESULT([$enable_valgrind])
+
+AS_IF([test "x$enable_valgrind" != "xno" -a "x$enable_memory_checking" != "xno"],
+    AC_MSG_ERROR([--enable-valgrind and --enable-memory-checking cannot be used at the same time])
+    )
+
```

Comment:
Optional: If either is enabled, we could also check whether `$with_gc = "gasman"` holds, and if not, print an error.